lib/ddtrace/contrib/rails/framework.rb

```diff
@@ -59,15 +59,19 @@ def self.configure(config)
 
           # set default database service details and store it in the configuration
           if defined?(::ActiveRecord)
-            adapter_name = ::ActiveRecord::Base.connection_config[:adapter]
-            adapter_name = Datadog::Contrib::Rails::Utils.normalize_vendor(adapter_name)
-            database_service = datadog_config.fetch(:default_database_service, adapter_name)
-            datadog_config[:default_database_service] = database_service
-            datadog_config[:tracer].set_service_info(
-              database_service,
-              adapter_name,
-              Datadog::Ext::AppTypes::DB
-            )
+            begin
+              conn_cfg = ::ActiveRecord::Base.connection_config()
+              adapter_name = Datadog::Contrib::Rails::Utils.normalize_vendor(conn_cfg[:adapter])
+              database_service = datadog_config.fetch(:default_database_service, adapter_name)
+              datadog_config[:default_database_service] = database_service
+              datadog_config[:tracer].set_service_info(
+                database_service,
+                adapter_name,
+                Datadog::Ext::AppTypes::DB
+              )
+            rescue StandardError => e
+              Datadog::Tracer.log.error("error while configuring database service: #{e}")
```

Comment:
If users have ``ActiveRecord`` but the connection isn't established (for any reasons) it seems a critical error. I'd use a different message like: ``database configuration not found; skipping ActiveRecord instrumentation`` (or something similar)